study-builder/tenants/testboston/rules/link-admin-accounts.js

```diff
@@ -0,0 +1,119 @@
+// This rule will look for admin account and link it to the Google social account. Account link is
+// determined by using the email. This is only executed for the admin client and only if the
+// accounts are not already linked.
+//
+// This was derived from Auth0's rule template from here:
+// https://github.com/auth0/rules/blob/b381ef4a1d5df16c1716ea066f46cff87b88f7f1/src/rules/link-users-by-email-with-metadata.js
+//
+// This assumes the rule runtime is Node 12, which is configured in Auth0 Dashboard Tenant Settings.
+function(user, context, callback) {
+  const _ = require('lodash');
+  const request = require('request');
+  const adminConnection = 'study-admin-authentication';
+  const googleConnection = 'google-oauth2';
+
+  // Note that client metadata key/values are all strings.
+  context.clientMetadata = context.clientMetadata || {};
+  const isAdminClient = context.clientMetadata.isAdminClient === 'true';
+
+  if (!isAdminClient || context.connection !== googleConnection) {
+    console.log('Either not admin client or not google connection, skipping account linking');
+    return callback(null, user, context);
+  }
+
+  const adminIdentity = user.identities.find(identity => identity.connection === adminConnection);
+  if (adminIdentity) {
+    console.log('Current user already has admin identity, meaning google identity has already been linked');
+    return callback(null, user, context);
+  }
+
+  request({
+    url: auth0.baseUrl + '/users',
```

Comment:
Auth0's docs recommend not using the endpoint during the authentication process as it is not immediately consistent, and references the `GET /api/v2/users-by-email` endpoint. Will the alternate endpoint serve an equivalent purpose for us here?

https://auth0.com/docs/users/user-search/retrieve-users-with-get-users-by-email-endpoint

Reply:
Good catch! I'll double-check the docs and switch to the `/users-by-email` endpoint.